### DISCUSSION TITLE
How can I strip down my parameter.json to collect only the parameters I want to retrieve?

### DISCUSSION THREAD

**Discussion by framps** [zivillian/ism7mqtt#138] [Q&A] [answered]
I successfully managed to create a parameter.json and run ism7mqtt and receive MQTT messages. I use telegraf to copy them into my local influxdb. As of now I copy all parameters.

I want to strip down my parameter.json so it just collects the parameters I want to see in grafana. I know ism7mqtt creates MQTT messages only if parameters changed. But nevertheless all parameters have to be queried. I want to reduce the query time and don't want to store all the data in my influxdb. I know I can select the parameters to copy into influxdb in telegraf but nevertheless all data will be queried by ism7mqtt.

How can I find out which parameter I have to keep in my parameter.json if I want to get for example Kesseltemperatur only? 

If I understand the algorithm I may create a script which accepts a parameter name and delivers 

```
 "ReadBusAddress": "0x8",
      "WriteBusAddress": "0x3",
      "DeviceTemplateId": 180000,
      "Parameter": [
          4711
      ]
```
and contribute it in a PR :wink:

**Comment by zivillian** [comment]
you can use the [ParameterTemplates.xml](https://github.com/zivillian/ism7mqtt/blob/29f9e43de8ef65711cad0c2a32905730e4cbf991/src/ism7mqtt/Resources/ParameterTemplates.xml) to look up your current parameters. Just search for the element with `PTID="<your id>"`.

The problem is, that the names are not unique - `Kesseltemperatur` is found 16 times and depending on your setup it will be one of those 16 ids.

If you want to know which parameters are supported for each device, you can look this up in [DeviceTemplates.xml](https://github.com/zivillian/ism7mqtt/blob/29f9e43de8ef65711cad0c2a32905730e4cbf991/src/ism7mqtt/Resources/DeviceTemplates.xml). For each device you can see a list of supported parameter ids.

**Reply by zivillian** [reply]
For your use case, it may be helpful to also have the parameter names in the json. Since this file is generated it should be easy to also add the parameter name. The remaining question is, how to migrate or still support the existing parameter.json files

**Comment by framps** [comment]
Thank you very much for your hints. 

I agree it would be helpful for my use case to have the parameter names included in the generated json. But it's actually not required because I just created a small bash script which accepts a search string and searches for all parameter names defined in `ParameterTemplates.xml` which match the search string and then it's checked whether the ptid is available in `parameter.json` .

If you find it useful I can contribute the script in a PR. Just let me know in which directory I should add the script and which name I should give the script.

Sample invocation and result:

```
framp@majestix:~/wolf$ findExistingISM7MQTTparameters.sh kessel parameter_full.json 
>>> "Kesseltemperatur": CGB-2 - 180005
>>> "Kesselübertemperatur Sammler": CGB-2 - 180068
>>> "Nachlaufzeit Kesselkreispumpen": CGB-2 - 180117
>>> "Kesselmaximaltemperatur HZ TV-max": CGB-2 - 180118
>>> "Kesselminimaltemperatur TK-min": CGB-2 - 180129
>>> "Kesselmaximaltemperatur TK-max": CGB-2 - 180130
>>> "Kesselübertemperatur bei Speicherladung": CGB-2 - 180131
>>> "Kesselsolltemperatur": CGB-2 - 180154
```

```
framp@majestix:~/wolf$ ./findExistingISM7MQTTparameters.sh wasser parameter_full.json 
>>> "Warmwassertemperatur": CGB-2 - 180006
>>> "Durchfluss Warmwasser": CGB-2 - 180010
>>> "Auslauftemperatur Warmwasser": CGB-2 - 180050
>>> "Obere Brennerleistung Warmwasser": CGB-2 - 180115
>>> "BUSCONFIG_Warmwasserkonfiguration": CGB-2 - 180140
>>> "Warmwassersolltemperatur": CGB-2 - 180155
>>> "Warmwasserminimaltemperatur": BM-2 - 220028
>>> "1x Warmwasser": BM-2 - 220042
>>> "Warmwasser Status": WWSystem_BM-2 - 350001
>>> "Warmwassersolltemperatur eingestellt": WWSystem_BM-2 - 350009
>>> "Warmwassersolltemperatur eingestellt": WWSystem_BM-2 - 350014
>>> "Warmwasserkreis Benamung": WWSystem_BM-2 - 350016

```

**Reply by framps** [reply]
> If you find it useful I can contribute the script in a PR

It's now 2 weeks ago I asked whether I should provide my script in a PR but didn't get any answer until now. It's fine with me if a bash script doesn't fit into your repo. I then will place the script in my bashScriptCollection repo and whoever is interested in this script can grab it from there.

**Comment by framps** [comment, marked as answer]
If somebody is interested in my script: It's located in my [bashScriptCollection](https://github.com/framps/bashScriptCollection)
